dali/operators/reader/nvdecoder/nvdecoder.cc

```diff
@@ -244,15 +244,15 @@ int NvDecoder::handle_decode_(CUVIDPICPARAMS* pic_params) {
   return kNvcuvid_success;
 }
 
-NvDecoder::MappedFrame::MappedFrame()
-    : MappedFrame(nullptr, nullptr, 0) {
-}
-
 NvDecoder::MappedFrame::MappedFrame(CUVIDPARSERDISPINFO* disp_info,
                                     CUvideodecoder decoder,
                                     CUstream stream)
     : disp_info{disp_info}, valid_{false}, decoder_(decoder), params_{0} {
 
+  if (!disp_info) {
+    DALI_FAIL("disp_info is nullptr.");
+  }
```

Comment:
That's a coding error, not a run-time error. If anything, it should be:
```suggestion
  assert(disp_info_);
```